### PR TITLE
Change signature of NewKeeper to directly accept a ConsensusHost

### PR DESCRIPTION
**Pull request by chatton**
## Description

closes: #6082

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

## Summary by CodeRabbit

- **Refactor**
	- Updated the initialization logic for certain components to use a `ConsensusHost` implementation instead of a `StakingKeeper`. This change aims to enhance clarity and functionality in the system's consensus mechanism.

### CODE REVIEW

File: modules/core/keeper/keeper_test.go [hunk 1]
```diff
@@ -5,17 +5,19 @@ import (
 	"testing"
 	"time"
 
+	ibctm "github.com/cosmos/ibc-go/v8/modules/light-clients/07-tendermint"
```
Comment (damiannolan) [inline modules/core/keeper/keeper_test.go:8]: compiler says no

**Review by damiannolan — APPROVED** [review]
ACK, think there's a compiler error in test file. We can also remove the expected keeper in 02-client/types for staking keeper. Otherwise LGTM!

**Review by crodriguezvega — APPROVED** [review]
Nice. :)

**Review by charleenfei — APPROVED** [review]
thank you @chatton!

**Pull request merged**